mitodl/ocw-studio
# Website.short_id for repo names

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #404 

#### What's this PR do?
- Adds `short_id` as a required, unique field to `Website`
- Adds the new field to the Website creation form
- Populates the field based on metadata during the OCW import process
- Updates the concourse pipeline definition to use this field as the git repo name
- Updates the `GithubApiWrapper` to use this field as the git repo name
- Backpopulates the field for existing websites

#### How should this be manually tested?
On the master branch, import some ocw sites if you don't have any yet (you will need some .env variables set: ):
  ```
  manage.py import_ocw_course_sites --bucket ocw-to-hugo-output-qa  --filter 15-00
  ```
Then switch to this branch and restart containers.  In django admin, check that all websites have a populated short_id field, and that the imported ocw sites have a short_id == `<course_number>-<semester>-<year>` in lowercase.

In particular, filter by "15-005" - there should be two ocw sites, one should have a `short_id=res.15-005-spring-2019` and the other should have `short_id=res.15-005-spring-2019-2`

Import some more sites on this branch, and make sure their short_id values are populated as above:
  ```
  manage.py import_ocw_course_sites --bucket ocw-to-hugo-output-qa  --filter 16-00
  ```

Create a new Website using the UI, make sure the short_id field appears on the form, validates correctly, and that the site is saved with the correct values.


#### Any background context you want to provide?
Since there are no production repos yet and just a handful of RC repos, decided to not create a management command for renaming them.

#### Screenshots (if appropriate)
<img width="689" alt="Screen Shot 2021-07-21 at 12 29 37 PM" src="https://user-images.githubusercontent.com/187676/126528557-24581771-a6de-4392-a7ca-6e738acd3bc5.png">

<img width="698" alt="Screen Shot 2021-07-21 at 12 31 02 PM" src="https://user-images.githubusercontent.com/187676/126528594-79af6118-7212-4f49-b8f2-284079da0044.png">
